SonarQube.MSBuild.Tasks/AttachBuildWrapper.cs

```diff
@@ -0,0 +1,187 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AttachBuildWrapper.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using SonarQube.Common;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace SonarQube.MSBuild.Tasks
+{
+    /// <summary>
+    /// MSBuild task to notify the BuildWrapper that the build of C++ project has started
+    /// </summary>
+    public class AttachBuildWrapper : Task
+    {
+        [DllImport("kernel32.dll")]
+        private static extern IntPtr GetModuleHandle(string lpModuleName);
+
+        // See https://jira.sonarsource.com/browse/CPP-1458 for the specification of the files contained in the C++ static resource zip
+        private const string BuildWrapperExeName = "build-wrapper-win-x86-64.exe";
+        private const string AttachedBinaryFileName32 = "interceptor32.dll";
+        private const string AttachedBinaryFileName64 = "interceptor64.dll";
+        private const string BuildWrapperSubDirName = "build-wrapper-win-x86";
+
+        private static readonly string[] RequiredFileNames = new string[] { BuildWrapperExeName, AttachedBinaryFileName32, AttachedBinaryFileName64 };
+
+        /// <summary>
+        /// The length of time to wait for the build wrapper to be launched succesfully
+        /// </summary>
+        private readonly int buildWrapperTimeoutInMs;
+
+        private const int DefaultBuildWrapperTimeoutInMs = 5000;
```

Comment:
The build wrapper should start within a second or so. This timeout is a fail-safe in case something goes wrong so we  don't cause the build to hang. If the attach request times out then the task will fail.